guarddog/analyzer/metadata/deceptive_author.py

```diff
@@ -0,0 +1,92 @@
+from abc import abstractmethod
+from typing import Optional
+import os
+from datetime import datetime, timedelta
+
+import requests
+
+from guarddog.analyzer.metadata.detector import Detector
+
+from .utils import extract_email_address_domain
+
+DISPOSABLE_EMAIL_DOMAINS_CACHE_LOCATION = os.environ.get(
+    "GUARDDOG_DISPOSABLE_EMAIL_DOMAINS_CACHE_LOCATION"
+)
+
+
+class DeceptiveAuthorDetector(Detector):
+    """This heuristic detects when an author is using a disposable email."""
+
+    MESSAGE_TEMPLATE = "An author was detected using a disposable email %s"
+
+    def __init__(self):
+        super().__init__(
+            name="deceptive_author",
+            description="This heuristic detects when an author is using a disposable email",
+        )
+
+    def get_email_addresses(self, package_info: dict) -> set[str]:
+        return set()
+
+    def _get_disposable_email_domains(self) -> list:
+        """
+        Gets the domains that are known to provide disposable e-mails
+
+        """
+
+        disposable_email_domains_url = \
```

Comment:
shouldn't we just bundle this within guarddog instead of attempting to fetch and cache it? I'm not sure I fully understand the current logic

Reply:
That's what this is doing.
This is a copy of existing code.

Line https://github.com/DataDog/guarddog/pull/374/files/ec37ad3ea4ed6428589159bbe190994b45448f5e#diff-7c851526ba215464ed17b5246b665267578958646496f493323e3016a585368fR55 makes use of getmtime to validate that this file is not older that 30 dias.  If so file gets redownloaded in line https://github.com/DataDog/guarddog/pull/374/files/ec37ad3ea4ed6428589159bbe190994b45448f5e#diff-7c851526ba215464ed17b5246b665267578958646496f493323e3016a585368fR65

Reply:
Maybe I can create encapsulate that code, since this way of proceeding is getting common

Reply:
right! sounds good then, perhaps just add a few comments

Reply:
There's  also this list: https://github.com/disposable-email-domains/disposable-email-domains available as a python package: https://pypi.org/project/disposable-email-domains/

Reply:
thanks @miketheman we liked the project and i implemented the changes